frontend/frontend_base.h

```diff
@@ -139,14 +139,15 @@ template <typename FrontendType> struct GenericFrontend
 
     Context *ctx;
     const FrontendType &impl;
+    const bool split_io;
     using mod_dat_t = typename FrontendType::ModuleDataType;
     using mod_port_dat_t = typename FrontendType::ModulePortDataType;
     using cell_dat_t = typename FrontendType::CellDataType;
     using netname_dat_t = typename FrontendType::NetnameDataType;
     using bitvector_t = typename FrontendType::BitVectorDataType;
 
     std::unordered_map<IdString, ModuleInfo> mods;
-    std::unordered_map<IdString, const mod_dat_t &> mod_refs;
+    std::unordered_map<IdString, const mod_dat_t> mod_refs;
```

Comment:
can you explain the reason for this change? it doesn't seem linked to the rest of the PR, even though I suspect it is a good idea

Reply:
Because `mod_dat_t` is simple `FrontendType::ModuleDataType`, the type is totally under the control of the `FrontendType`.  The json frontend already marks `ModuleDataType` as a reference, but the FPGA interchange frontend is instead using a lightwight value type.

See https://github.com/YosysHQ/nextpnr/blob/c956cae8244c094783edc7101fd0ca542c24e55b/frontend/json_frontend.cc#L37

If you want I can separate this change from this PR.

Reply:
That makes sense, I'm happy with that being part of this PR.